app/controllers/spree/retail/shopify/hooks_controller.rb

```diff
@@ -0,0 +1,28 @@
+module Spree
+  module Retail
+    module Shopify
+      class HooksController < ApplicationController
+        before_filter :verify_request_authenticity
+
+        private
+
+        def verify_request_authenticity
+          data = request.body.read
+
+          head :unauthorized unless request_authentified?(data, Spree::Retail::Config.shopify_webhook_shared_secret)
+        end
+
+        def request_authentified?(data, secret)
```

Comment:
What do you think of encapsulating this logic into a nice `ShopifyPayload` object with an elegant interface (e.g. `#valid?`, etc.)?

Additionally, I think `authenticated` is the word you're looking for here. 😄